File: app/src/main/java/com/weiqi/ui/screens/SettingsScreen.kt
package com.weiqi.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.weiqi.ui.components.ZenCard

@Composable
fun SettingsScreen() {
    Column(
        Modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background)
            .verticalScroll(rememberScrollState())
            .padding(24.dp)
    ) {
        Text("Settings",
            style = MaterialTheme.typography.headlineMedium,
            fontWeight = FontWeight.SemiBold)
        Spacer(Modifier.height(16.dp))
        ZenCard(Modifier.fillMaxWidth()) {
            Column {
                SettingRow("Board theme", "Kaya")
                Spacer(Modifier.height(12.dp))
                SettingRow("Coordinates", "Off")
                Spacer(Modifier.height(12.dp))
                SettingRow("Sound", "On")
                Spacer(Modifier.height(12.dp))
                SettingRow("Beginner hints", "On")
            }
        }
    }
}

@Composable
private fun SettingRow(label: String, value: String) {
    androidx.compose.foundation.layout.Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = androidx.compose.foundation.layout.Arrangement.SpaceBetween
    ) {
        Text(label, style = MaterialTheme.typography.bodyMedium)
        Text(value,
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant)
    }
}

File: lib/src/ui/screens/settings_screen.dart
import 'package:flutter/material.dart';

import '../../data/settings_store.dart';
import '../components/zen_components.dart';

class SettingsScreen extends StatefulWidget {
  final SettingsStore store;
  const SettingsScreen({super.key, required this.store});

  @override
  State<SettingsScreen> createState() => _SettingsScreenState();
}

class _SettingsScreenState extends State<SettingsScreen> {
  @override
  void initState() {
    super.initState();
    widget.store.addListener(_onChange);
  }

  @override
  void dispose() {
    widget.store.removeListener(_onChange);
    super.dispose();
  }

  void _onChange() => setState(() {});

  @override
  Widget build(BuildContext context) {
    final scheme = Theme.of(context).colorScheme;
    final text = Theme.of(context).textTheme;
    final s = widget.store.value;

    return SingleChildScrollView(
      padding: const EdgeInsets.all(20),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          Text('Settings',
              style:
                  text.headlineMedium?.copyWith(fontWeight: FontWeight.w600)),
          const SizedBox(height: 12),
          ZenCard(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.stretch,
              children: [
                Text('BOARD THEME',
                    style: text.labelSmall
                        ?.copyWith(color: scheme.onSurfaceVariant)),
                const SizedBox(height: 8),
                _themeRow([
                  (BoardThemeKind.classicWood, 'Wood'),
                  (BoardThemeKind.minimalPaper, 'Paper'),
                ], s),
                const SizedBox(height: 8),
                _themeRow([
                  (BoardThemeKind.darkSlate, 'Slate'),
                  (BoardThemeKind.highContrast, 'High Contrast'),
                ], s),
              ],
            ),
          ),
          const SizedBox(height: 12),
          ZenCard(
            child: Column(
              children: [
                _ToggleRow(
                  label: 'Coordinates',
                  checked: s.showCoordinates,
                  onToggle: () => widget.store.update(
                      (v) => v.copyWith(showCoordinates: !v.showCoordinates)),
                ),
                _ToggleRow(
                  label: 'Move numbers in review',
                  checked: s.showMoveNumbers,
                  onToggle: () => widget.store.update(
                      (v) => v.copyWith(showMoveNumbers: !v.showMoveNumbers)),
                ),
                _ToggleRow(
                  label: 'Beginner hints',
                  checked: s.beginnerHints,
                  onToggle: () => widget.store.update(
                      (v) => v.copyWith(beginnerHints: !v.beginnerHints)),
                ),
              ],
            ),
          ),
          const SizedBox(height: 16),
          Text(
            'Weiqi · Local MVP',
            style: text.labelSmall?.copyWith(color: scheme.onSurfaceVariant),
          ),
        ],
      ),
    );
  }

  Widget _themeRow(List<(BoardThemeKind, String)> opts, AppSettings s) {
    return Row(
      children: [
        for (var i = 0; i < opts.length; i++) ...[
          Expanded(
            child:
                _themeChip(opts[i].$1, opts[i].$2, s.boardTheme == opts[i].$1),
          ),
          if (i != opts.length - 1) const SizedBox(width: 8),
        ]
      ],
    );
  }

  Widget _themeChip(BoardThemeKind theme, String label, bool selected) {
    final scheme = Theme.of(context).colorScheme;
    return FilterChip(
      selected: selected,
      onSelected: (_) =>
          widget.store.update((v) => v.copyWith(boardTheme: theme)),
      label: Text(label,
          style:
              TextStyle(color: selected ? scheme.onPrimary : scheme.onSurface)),
      backgroundColor: scheme.surfaceContainerHigh,
      selectedColor: scheme.primary,
      checkmarkColor: scheme.onPrimary,
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(14)),
      side: BorderSide.none,
    );
  }
}

class _ToggleRow extends StatelessWidget {
  final String label;
  final bool checked;
  final VoidCallback onToggle;
  const _ToggleRow(
      {required this.label, required this.checked, required this.onToggle});

  @override
  Widget build(BuildContext context) {
    final text = Theme.of(context).textTheme;
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 4),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: [
          Text(label, style: text.bodyMedium),
          Switch(value: checked, onChanged: (_) => onToggle()),
        ],
      ),
    );
  }
}
